src/config/source.rs

```diff
@@ -131,6 +131,8 @@ pub struct SourceContext {
     /// Given a source can expose multiple [`SourceOutput`] channels, the ID is tied to the identifier of
     /// that `SourceOutput`.
     pub schema_definitions: HashMap<Option<String>, schema::Definition>,
+
+    pub extra_context: ExtraContext,
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏non-blocking. Would it be helpful to add a doc comment describing what this can be or is used for?

Reply:
Sure. The original line in the `SinkContext` didn't have a comment, but that doesn't mean there shouldn't be one.